src/main/java/org/gelecekbilimde/scienceplatform/auth/model/RoleSelfApplicationFilter.java

```diff
@@ -0,0 +1,50 @@
+package org.gelecekbilimde.scienceplatform.auth.model;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+import org.gelecekbilimde.scienceplatform.auth.model.entity.RoleApplicationEntity;
+import org.gelecekbilimde.scienceplatform.auth.model.enums.RoleApplicationStatus;
+import org.gelecekbilimde.scienceplatform.common.model.BaseFilter;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.util.CollectionUtils;
+
+import java.util.Set;
+
+@Getter
+@Setter
+@Builder
+public class RoleSelfApplicationFilter implements BaseFilter {
+
+	private Set<RoleApplicationStatus> statuses;
+	private User user;
+
+	@Getter
+	@Setter
+	@Builder
+	public static class User {
+		private String id;
+	}
+
+
+	@Override
+	public Specification<RoleApplicationEntity> toSpecification() {
+
+		Specification<RoleApplicationEntity> specification = (root, query, criteriaBuilder) ->
+			criteriaBuilder.equal(root.join("user").get("id"), this.user.getId());
```

Comment:
@coderabbitai Since we set the user's ID information ourselves, we have full control. The user cannot intervene from the outside. Therefore, there is no such thing as being null.